src/Acuminator/Acuminator.Utilities/DiagnosticSuppression/SuppressionFile.cs

```diff
@@ -126,28 +126,32 @@ private static XElement ElementFromMessage(SuppressMessage message)
 				new XElement(SyntaxNodeElement, message.SyntaxNode));
 		}
 
-		private static SuppressMessage MessageFromElement(XElement element)
-		{
-			var id = element.Attribute(IdAttribute).Value;
-			var targetElement = element.Element(TargetElement);
-			var target = targetElement.Value;
-			var syntaxNode = element.Element(SyntaxNodeElement).Value;
-
-			return new SuppressMessage(id, target, syntaxNode);
-		}
-
 		public static HashSet<SuppressMessage> LoadMessages(ISuppressionFileSystemService fileSystemService, string path)
 		{
-			var document = fileSystemService.Load(path);
+			XDocument document;
+
+			lock (fileSystemService)
```

Comment:
@SENya1990 why do we need a look here?

Reply:
@metallmonster because the VS throws exception about access to file from another thread.

Reply:
@metallmonster because the VS throws exception about access to file from another thread.